### DISCUSSION TITLE
Caching Rain Radar Images

### DISCUSSION THREAD

**Discussion by roly151** [JaffaKetchup/flutter_map_tile_caching#140] [Q&A] [answered]
I am using this package to cache rain radar images obtained from Rain Viewer. I use a timer to update the TileLayer url ('https://tilecache.rainviewer.com${weatherController.rainViewerData[currentRadarIndex.value].path}/256/{z}/{x}/{y}/4/1_1.png'). For the most part this works well once I have 'cycled through' the images and they are showing for the second time. However, the rainviewer api is very slow so the 'animation' does not really move until the images are cached. So I thought about preloading or downloading the rain radar images. 

I have therefore created a rectangle region, and following the docs, need to make that region downloadable, with something like this:

`final downloadable = region.toDownloadable(
minZoom: 3,
maxZoom: 10, 
options: TileLayer(
        urlTemplate:
          'https://tilecache.rainviewer.com${weatherController.rainViewerData[currentRadarIndex.value].path}/256/{z}/{x}/{y}/4/1_1.png',
        tileDisplay: const TileDisplay.instantaneous(),
      ),
    );`

But how do I deal with the different url's? (I'm only a beginner with flutter, so not sure how to proceed). Any help is appreciated.

**Comment by JaffaKetchup** [comment]
Hi @roly151,
You seem to have started on the right track. All you need to do is ensure you use the same `urlTemplate` that you downloaded with when it comes to the `TileLayer` in the actual map. Essentially, the two `TileLayer`s (one for FMTC and one for FM) should match (at least for the URL).
Tiles are retrieved from the cache by URL, so you can have multiple different URLs within the same store without an issue.
Hopefully that makes sense!

**Reply by roly151** [reply]
Hi, thanks for the reply. I have the same url in FM. 

My question is do I need to cycle through all the different urls and download the tiles for the region for each of them? 

Thanks.

**Reply by JaffaKetchup** [reply, marked as answer]
Ah yes, you'll need to start a different download for every URL.
